pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1656,6 +1660,7 @@ func (l *LibvirtDomainManager) ListAllDomains() ([]*api.Domain, error) {
 			return list, err
 		}
 		domain.SetState(util.ConvState(status), util.ConvReason(status, reason))
+		domain.ObjectMeta.UID = l.uid
```

Comment:
@EdDev Hi. Can you please elaborate on the reason you chose to set the `UID` here instead of doing so in the [converter](https://github.com/kubevirt/kubevirt/blob/82a415c11074509b13a304d15c4119ec691c31c4/pkg/virt-launcher/virtwrap/converter/converter.go#L1158) function? I see that the domain metadata is also initialized in the converter, and the KV metadata. I think it is early enough so the virt-handler cache lister will receive the UID

Reply:
We do not need it in the converter.
The converter takes a VMI spec and translates it to a domain object, which is then used to create a libvirt domain (dom).
In this direction, we do not need the UID at all.
This domain, is not preserved anywhere except in libvirt as the domxml.

We need to have the UID available when we create a domain object that needs to be sent to virt-handler.
There are several methods that create a domain object in that flow:
- [NewDomainFromName](https://github.com/kubevirt/kubevirt/pull/8042/files#diff-16ad880bd691a5496ed2dcad8b5b98a8a50f2ae5f1fac21a01579adde1447de6R441)
- [NewDomain](https://github.com/kubevirt/kubevirt/pull/8042/files#diff-16ad880bd691a5496ed2dcad8b5b98a8a50f2ae5f1fac21a01579adde1447de6R429)
  - As we no longer have the UID available in the libvirt domxml (dom), it is added [when it is available](https://github.com/kubevirt/kubevirt/pull/8042/files#diff-98b8339212e612cd92ce33649e492336892a3ed649f368b181aad9aa2854138cR1663).